lms/djangoapps/mobile_api/users/serializers.py

```diff
@@ -34,10 +36,17 @@ def to_native(self, course_overview):
                 kwargs={'course_id': course_id},
                 request=request
             )
+            discussion_url = reverse(
+                'discussion_course',
+                kwargs={'course_id': course_id},
+                request=request
+            ) if 'discussion' in ast.literal_eval(course_overview.tabs) and DiscussionTab.is_enabled(course_overview) \
```

Comment:
Calling `DiscussionTab.is_enabled` will result in a performance regression since it re-checks enrollment and course access.  Instead, the `DiscussionTab.is_enabled` method needs to be refactored so only the necessary code needed by this call is run.